lib/simulator-xcode-6.js

```diff
@@ -768,6 +768,36 @@ class SimulatorXcode6 extends EventEmitter {
   }
 
   /**
+   * Perform Simulator caches cleanup.
+   *
+   * @param {?array<string>} folderNames - The names of Caches subfolders to be cleaned.
+   *   Non-accessible/non-existing subfolders will be skipped.
+   *   All existing subfolders under Caches will be deleted if this parameter is omitted.
+   * @returns {SimulatorXcode6} This instance for chaining.
+   */
+  async clearCaches (...folderNames) {
+    const cachesRoot = path.resolve(this.getDir(), 'Library', 'Caches');
+    if (!await fs.hasAccess(cachesRoot)) {
+      log.debug(`Caches root at '${cachesRoot}' does not exist or is not accessible. Nothing to do there`);
+      return this;
+    }
+
+    let itemsToRemove = folderNames.length ?
+      folderNames.map((x) => path.resolve(cachesRoot, x)) :
+      (await fs.readdir(cachesRoot)).map((x) => path.resolve(cachesRoot, x));
+    itemsToRemove = await B.filter(itemsToRemove, (x) => fs.hasAccess(x));
+    itemsToRemove = await B.filter(itemsToRemove, async (x) => (await fs.stat(x)).isDirectory());
```

Comment:
These could all be chained if you wanted (since everything returns a Promise.

Reply:
Which is more likely to filter more things, in the normal case? The order might need adjusting if there are more non-directories than directories that cannot be accessed.

Reply:
moved isDirectory check to the top